### DISCUSSION TITLE
How to export materials referencing textures with relative paths?

### DISCUSSION THREAD

**Discussion by TechArtistG** [Autodesk/maya-usd#1068] [Q&A]
Hi all, we have a Maya scene setup where materials have relative paths to textures.  When we export to USD I'd export the USD shader graph to just keep the exact path we have specified in Maya, but it resolves out to absolute paths.  We've tried this with/without using Maya projects and no change.

Is there a way to export materials from Maya with relative texture paths to the resultant USD file?

**Comment by JGamache-autodesk** [comment]
Hello! We would need more information. For `.usda` and `.usd` files the path are relative since #686 and #743. For `.usdz` files, @dgovil fixed the remaining issue in #1028 

Can you tell us which version of MayaUSD you are using and what are the steps taken to reproduce the issue? That would help a lot.
Thanks.

**Comment by womanyoyoyo-adsk** [comment]
Hi, we have this now in MayaUSD 0.26.0.

**Comment by Ark-Dem** [comment]
When dealing with Arnold aiImage there seem to be two approaches to paths:
(1) token inputs:filename = @./color.exr@
It works fine, but LookdevX writes an error that this attribute should be of a string type as defined in Arnold description...
(2) string inputs:filename = "[envvar]/color.exr"
It works fine in LookdevX, no errors, but no relative-to-usd paths anymore (I'm substituting it with environment variable).

What is the plan for this one? Do you plan to make (1) work in LookdevX without errors or make (2) work with relative-to-usd paths ("./color.exr" or @./color.exr@)?

**Comment by sharktacos** [comment]
If this is an Arnold render in Maya, make sure that in the Render Settings/System/Search Paths that "absolute texture paths" is not checked (it is by default).
